tasks/Task.cpp

```diff
@@ -25,6 +25,25 @@ Task::~Task()
 
 bool Task::configureHook()
 {
+    /**
+     * Apply the camera parameters
+     */
+    underwater_camera_simulation::CameraParams params = _camera_params.get();
+
+    /**
+     * set the viewport size the scene size of the camera
+     */
+     if (params.width <=0 || params.height <=0)
+     {
+         std::invalid_argument("The camera width and height must be greater than zero");
+     }
+     else
+     {
+         _width.set(params.width);
```

Comment:
Oh this is bad ... same configuration parameter in two different places ... Hugh.

What is the point of the width/height in CameraParams if it does not set anything ? My understanding is that the actual width/height is really set by the width and height properties, so I would just remove it from the params structure.